hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java

```diff
@@ -136,19 +136,22 @@ private void createCompactionExecutors() {
     Preconditions.checkArgument(largeThreads > 0 && smallThreads > 0);
 
     final String n = Thread.currentThread().getName();
-
     StealJobQueue<Runnable> stealJobQueue = new StealJobQueue<Runnable>(COMPARATOR);
-    this.longCompactions = new ThreadPoolExecutor(largeThreads, largeThreads, 60,
-        TimeUnit.SECONDS, stealJobQueue,
-        new ThreadFactoryBuilder()
-            .setNameFormat(n + "-longCompactions-" + System.currentTimeMillis())
-            .setDaemon(true).build());
+
+    AtomicInteger longCompactionThreadCounter = new AtomicInteger(0);
+    this.longCompactions =
+        new ThreadPoolExecutor(largeThreads, largeThreads, 60, TimeUnit.SECONDS, stealJobQueue,
+            new ThreadFactoryBuilder().setNameFormat(n + "-longCompactions-"
+                + longCompactionThreadCounter.getAndIncrement() + "-" + System.currentTimeMillis())
+                .setDaemon(true).build());
     this.longCompactions.setRejectedExecutionHandler(new Rejection());
     this.longCompactions.prestartAllCoreThreads();
-    this.shortCompactions = new ThreadPoolExecutor(smallThreads, smallThreads, 60,
-        TimeUnit.SECONDS, stealJobQueue.getStealFromQueue(),
-        new ThreadFactoryBuilder()
-            .setNameFormat(n + "-shortCompactions-" + System.currentTimeMillis())
+
+    AtomicInteger shortCompactionThreadCounter = new AtomicInteger(0);
+    this.shortCompactions = new ThreadPoolExecutor(smallThreads, smallThreads, 60, TimeUnit.SECONDS,
+        stealJobQueue.getStealFromQueue(),
+        new ThreadFactoryBuilder().setNameFormat(n + "-shortCompactions-"
+            + shortCompactionThreadCounter.getAndIncrement() + "-" + System.currentTimeMillis())
```

Comment:
> Does the 'n' have a thread number in it? 

No, it will be just RS name/ip-port.

May be you can have a look into below discussion,
https://issues.apache.org/jira/browse/HBASE-22930?focusedCommentId=16922548&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16922548